src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.c

```diff
@@ -100,205 +108,222 @@ void func_80AC9FE4(EnTimeTag* this, PlayState* play) {
             ActorCutscene_SetIntentToPlay(this->actor.cutscene);
         } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
             ActorCutscene_Start(this->actor.cutscene, &this->actor);
-            this->actor.home.rot.x = 0;
+            ENTIMETAG_GET_DIARY_TIMER(&this->actor) = 0;
         } else {
             ActorCutscene_SetIntentToPlay(this->actor.cutscene);
         }
     }
 }
 
-void func_80ACA5F8(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_Diary_Wait(EnTimeTag* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
         if (gSaveContext.save.playerForm == PLAYER_FORM_ZORA) {
-            if (ENTIMETAG_GET_SWITCHFLAG(&this->actor) == 1) {
+            if (ENTIMETAG_GET_DIARY_TYPE(&this->actor) == DIARY_LULU) {
                 Message_StartTextbox(play, 0x101C, &this->actor);
             } else {
+                // DIARY_MIKAU
                 Message_StartTextbox(play, 0x122B, &this->actor);
             }
-            this->actor.home.rot.x = 1;
+            ENTIMETAG_GET_DIARY_TIMER(&this->actor) = 1;
         } else {
+            // unable to read Zora script
             Message_StartTextbox(play, 0x122A, &this->actor);
+
+            //! FAKE:
             if (0) {}
+
             ((EnElf*)GET_PLAYER(play)->tatlActor)->unk_264 |= 4;
             Actor_ChangeFocus(&this->actor, play, GET_PLAYER(play)->tatlActor);
-            this->actor.home.rot.x = 0;
+            ENTIMETAG_GET_DIARY_TIMER(&this->actor) = 0;
         }
-        this->actionFunc = func_80ACA418;
+        this->actionFunc = EnTimeTag_Diary_Cutscene;
     } else if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, play)) {
         func_800B8614(&this->actor, play, 110.0f);
     }
 }
 
-void func_80ACA714(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_KickOut_DoNothing(EnTimeTag* this, PlayState* play) {
 }
 
-void func_80ACA724(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_KickOut_Transition(EnTimeTag* this, PlayState* play) {
     if (Message_GetState(&play->msgCtx) == TEXT_STATE_5) {
-        play->nextEntrance = play->setupExitList[ENTIMETAG_GET_1F(&this->actor)];
+        play->nextEntrance = play->setupExitList[ENTIMETAG_GET_KICKOUT_EXIT_INDEX(&this->actor)];
         play->transitionTrigger = TRANS_TRIGGER_START;
-        if (!ENTIMETAG_GET_E000(&this->actor)) {
+        if (ENTIMETAG_GET_TYPE(&this->actor) == TIMETAG_KICKOUT_DOOR) {
             Actor_PlaySfxAtPos(&this->actor, NA_SE_OC_DOOR_OPEN);
         }
-        this->actionFunc = func_80ACA714;
+        this->actionFunc = EnTimeTag_KickOut_DoNothing;
     }
 }
 
-void func_80ACA7C4(EnTimeTag* this, PlayState* play) {
+/**
+ * Setup a request to kick out, but wait for an event to complete first
+ */
+void EnTimeTag_KickOut_WaitForEvent(EnTimeTag* this, PlayState* play) {
     if (!(gSaveContext.save.weekEventReg[63] & 1) && !(gSaveContext.save.weekEventReg[63] & 2)) {
```

Comment:
Is anything known about these two? Presumably it's one particular event with 2 attached flags.

Reply:
Yeah, both are entirely designed around being kicked out, but sometimes an event is happening at the specific time you're suppose to get kicked out (like a minigame), so this delays it.

For example, if the shooting gallery minigame is playing past the time you're suppose to kick out, you would never get kicked out normally due to it being a specific time. But using these two `weekEventReg`, you can "store" the fact that the "kick-out" event happened, and you can get kicked out once the minigame is fished.

`gSaveContext.save.weekEventReg[63] |= 1` -> this is set by the actor to request that if the kick-out time is passed, to store that kickout time.

`gSaveContext.save.weekEventReg[63] |= 2` -> this is set by time-tag to indicate that the kick-out time has passed and is now stored

Now with both of these flags set, it is now up to the actor to unset both of these, which will trigger the "delayed kick-out" after the requested time

Reply:
So
`WEEKEVENTREG_63_01` -> `WEEKEVENTREG_REQUEST_KICKOUT_DELAY` or `WEEKEVENTREG_KICKOUT_REQUEST_STORED`

`WEEKEVENTREG_63_02` -> `WEEKEVENTREG_KICKOUT_HAS_BEEN_DELAYED` or `WEEKEVENTREG_KICKOUT_IS_STORED`

as some ideas